File: codingame/skynet-the-chasm.rb
#!/usr/bin/ruby
STDOUT.sync = true
R = gets.to_i
G = gets.to_i
L = gets.to_i

loop{
	s = gets.to_i
	x = gets.to_i
	if s+x-1>=R
		puts 'JUMP'
		break
	elsif s<=G
		puts 'SPEED'
	elsif s>G+1
		puts 'SLOW'
	else
		puts 'WAIT'
	end
}
loop{puts 'SLOW'}

File: codingame/skynet-the-chasm.js
var R = parseInt(readline());
var G = parseInt(readline());
var L = parseInt(readline());

for(;;){
	var s = parseInt(readline());
	var x = parseInt(readline());
	if(s+x-1>=R){
		print('JUMP');
		break;
	}else if(s<=G){
		print('SPEED');
	}else if(s>G+1){
		print('SLOW');
	}else{
		print('WAIT');
	}
}
for(;;)print('SLOW');